static/js/robot.js

```diff
@@ -91,30 +349,159 @@ class RobotClass {
             let j = Math.sin(theta) * rt;
             dx = i * Math.sin(radian) + j * Math.cos(radian);
             dy = i * Math.cos(radian) + j * Math.sin(radian);
-            this.dir= (this.dir + theta*180/Math.PI) % 360;
+            this.dir = (this.dir + theta*180/Math.PI) % 360;
+          }
+          const ogX = this.X;
+          const ogY = this.Y;
+          const ogltheta = this.ltheta;
+          const ogrtheta = this.rtheta;
+
+          this.X = Math.max(Math.min(this.X + dx, this.MaxX), 0);
+          this.Y = Math.max(Math.min(this.Y + dy, this.MaxY), 0);
+          this.ltheta = (this.Wl * 5 + this.ltheta) % 360;
+          this.rtheta = (this.Wr * 5 + this.rtheta) % 360;
+
+          this.updateCorners(this.X - ogX, this.Y - ogY, ogDir);
```

Comment:
Instead of changing the corners back and forth if there is a collision, change the `intersectOne()` function to take in the new corners and check if there is a collision. Then, only update the corners if there is no collision.